sdk/core/core/inc/az_span.h

```diff
@@ -20,7 +21,9 @@
 #include <_az_cfg_prefix.h>
 
 /**
- * An immutable span of bytes (octets).
+ * az_span is a "view" over a byte buffer. It contains a pointer to the
+ * start of the byte buffer, the buffer's capacity; and the length (number of
+ * bytes) in use from the start of the buffer.
```

Comment:
I'm tempted to not do this because, on smaller MCUs, we may switch to 16-bit integers and we really haven't resolved the signed vs unsigned issue yet either. If we say something here, it is likely to be wrong later.

Reply:
Aren't we already encoding some of this behavior in the public surface area? For instance, the length getter returns an int32. Maybe, we shouldn't shy away from documenting the current behavior as is. If we were to change it while in preview, we can change the docs then too.

The only reason I can see us not documenting such detail is that if we are able to change it in subsequent releases without affecting the public surface area in any significant observable way. I am not sure if that's feasible.

Reply:
We may change the types during the preview period. We were having debates on this and so it was in flux. The debates stopped and so this may just make it all the way to GA. Then we can doc the int types.